src/corelibs/U2View/src/ov_msa/Export/MSAImageExportTask.cpp

```diff
@@ -306,16 +306,9 @@ void MSAImageExportController::initSettingsWidget() {
     CHECK(!selection.isEmpty(), );
     msaSettings.region = U2Region(selection.x(), selection.width());
     msaSettings.seqIdx.clear();
-    if (!ui->isVirtualOrderMode()) {
-        for (qint64 i = selection.y(); i < selection.height() + selection.y(); i++) {
-            msaSettings.seqIdx.append(i);
-        }
-    } else {
-        MaCollapseModel *model = ui->getCollapseModel();
-        SAFE_POINT(model != NULL, tr("MSA Collapsible Model is NULL"), );
-        for (qint64 i = selection.y(); i < selection.height() + selection.y(); i++) {
-            msaSettings.seqIdx.append(model->getMaRowIndexByViewRowIndex(i));
-        }
+    MaCollapseModel *model = ui->getCollapseModel();
```

Comment:
Can `ui` and `model` be `nullptr` here and elsewhere?

Reply:
`ui` or `model` can't be null in MSAImageExportController or in any other place where it is expected to be passed/stored.

However there is a design issue when some widgets are cached in async tasks without smart pointers, but this is a different story.